File: frontend/src/types/industry.ts
export interface IndustryRotationItem {
  industry_code: string
  industry_name: string
  last_date: string
  latest_close: number | null
  return_1m: number
  return_3m: number
  rank_1m: number
  rank_3m: number
  rotation_type: string
  generated_at: string
}

export interface IndustryRotationResponse {
  source: string
  data_date: string
  items: IndustryRotationItem[]
  total: number
}

File: backend/app/schemas/industry.py
from pydantic import BaseModel


class IndustryRotationItem(BaseModel):
    industry_code: str
    industry_name: str
    last_date: str
    latest_close: float | None = None
    return_1m: float
    return_3m: float
    rank_1m: int
    rank_3m: int
    rotation_type: str
    generated_at: str


class IndustryRotationResponse(BaseModel):
    warning: str | None = None
    source: str
    data_date: str
    items: list[IndustryRotationItem]
    total: int


class IndustryDatesResponse(BaseModel):
    items: list[str]
